### PR TITLE
feat: add resource_deployment_environment as fast filter in traces page

### PR DESCRIPTION
**Pull request by ahmadshaheer**
### Summary
Add Environment as a fast filter in trace explorer page. 

#### Related Issues / PR's
part of https://github.com/SigNoz/signoz/issues/3184

#### Screenshots
image

NA

### CODE REVIEW

**Review by rahulkeswani101 — DISMISSED** [review]

**Review by vikrantgupta25 — DISMISSED** [review]

File: frontend/src/pages/TracesExplorer/Filter/filterUtils.ts [hunk 1]
```diff
@@ -8,10 +9,14 @@ import {
 import { TagFilterItem } from 'types/api/queryBuilder/queryBuilderData';
 import { DataSource } from 'types/common/queryBuilder';
 
-export const AllTraceFilterKeyValue = {
+const RESOURCE_DEPLOYMENT_ENVIRONMENT = convertMetricKeyToTrace(
+	'resource_deployment_environment',
+);
```
Comment (srikanthccv) [inline frontend/src/pages/TracesExplorer/Filter/filterUtils.ts:14]: You don't need to take `resource_deployment_environment` and convert it. You can directly use `deployment.environment`.  Conversion is only required when we go from metrics to traces/logs.

Comment (ahmadshaheer) [reply frontend/src/pages/TracesExplorer/Filter/filterUtils.ts]: Done

Comment (ahmadshaheer) [reply frontend/src/pages/TracesExplorer/Filter/filterUtils.ts]: Done, @srikanthccv

**Review by rahulkeswani101 — APPROVED** [review]

**Comment by ankitnayan** [comment]
keep the `Environment` section expanded by default like `Service Name` or `Status`? I see it collapsed by default

**Review by ankitnayan — DISMISSED** [review]

**Comment by ahmadshaheer** [comment]
> keep the `Environment` section expanded by default like `Service Name` or `Status`? I see it collapsed by default

Done, @ankitnayan .

**Comment by ankitnayan** [comment]
@SagarRajput-7 I see the QB to fast filters mapping works only for `IN` operator and not for `=` operator. Is this expected?

**Comment by SagarRajput-7** [comment]
> @SagarRajput-7 I see the QB to fast filters mapping works only for `IN` operator and not for `=` operator. Is this expected?

Yes at that time, this was difficult to cover. So we decided to take it later. Also, @vikrantgupta25 and I discussed covering this with the new Logs filters, as he made it in a generic sense and over our calls, we considered the Traces filter's requirement in mind.

FYI - @vikrantgupta25

**Comment by ankitnayan** [comment]
> > @SagarRajput-7 I see the QB to fast filters mapping works only for `IN` operator and not for `=` operator. Is this expected?
> 
> Yes at that time, this was difficult to cover. So we decided to take it later. Also, @vikrantgupta25 and I discussed covering this with the new Logs filters, as he made it in a generic sense and over our calls, we considered the Traces filter's requirement in mind.
> 
> FYI - @vikrantgupta25

please create an issue for this. Atleast someday we will look into this

**Review by ankitnayan — DISMISSED** [review]

**Comment by SagarRajput-7** [comment]
Here - https://github.com/SigNoz/signoz/issues/5900

**Review by SagarRajput-7 — DISMISSED** [review]

**Review by YounixM — DISMISSED** [review]

**Comment by ankitnayan** [comment]
@ahmadshaheer the build needs to be fixed for this to merge, right?

**Pull request merged**